docs/build/wasm/ask_contracts.md

```diff
@@ -1,82 +1,221 @@
 # Ask! Smart Contracts
 
+:::caution
+
+Ask! eDSL has many [limitations and issues](https://github.com/ask-lang/ask/issues/)
+which are actively worked upon but at this moment it is not recommended
+for **PRODUCTION** environment. Please consider [ink!](dsls#ink) if you are building a contract for production.
+
+:::
+
 This guide will support you to set up your local environment and deploy a simple Ask! contract on our testnet.
 
 ---
 
+## What will we do?
+
+We will setup the local environment for developing ask! smart contract and deploys it Shibuya Testnet.
+
 ## What is Ask!?
 
-Ask! is a framework for AssemblyScript developers to write WASM smart contracts for `pallet-contracts`. Its syntax is similar to TypeScript. [Current project](https://polkadot.polkassembly.io/post/949) is funded by Polkadot treasury. Please keep in mind that Ask! is still under development and no release is available at the moment.
+Ask! is a framework for AssemblyScript developers to write WASM smart contracts for `pallet-contracts`. Its syntax is similar to TypeScript. [Current project](https://polkadot.polkassembly.io/post/949) is funded by Polkadot treasury. Please keep in mind that Ask! is still under development and can be quite buggy and restrictive.
 
-## Ask! Environment Setup
+## Prerequisites
 
-Start with executing the following command: 
+This tutorial targets developers who are new to ask! and AssemblyScript.
+Prior knowledge of using/setting up typescript/javascript project is helpful but not needed.
+
+## Setup Environment
+
+#### Install yarn package manager.
+
+We will be using `yarn` package manager to manage our ask! project. For installation instructions [read here](https://classic.yarnpkg.com/lang/en/docs/install)
```

Comment:
@ashutoshvarma - does it matter which version of yarn is used (I assume it does since Ask repo needs to be cloned)? You explicitly linked the version (classic) needed, but if the version matters maybe it is a good idea to to add a small note (for the complete begginers) that they need v1/classic and not v2

Reply:
No it does not for this flipper contract. We are not cloning/patching ask! in any way for flipper.

btw it does not matter in most cases, as projects that uses yarnV2 have `"packageManager": "yarn@3.3.1"` (like [here](https://github.com/ashutoshvarma/psp34-contract/blob/4bf323c291eae5ddbadfae122c198a3a04df7f42/package.json#L4)) line in their `package.json` which yarnV1 understands and automatically fetchs yarnV2 (if not installed) and use it instead.